### PR TITLE
Week 3 check-in

### PR DESCRIPTION
**Pull request by tmikeschu**
This review felt pretty straightforward to me, and I feel comfortable with the content. I really enjoy solidifying the concepts each week with these. Thanks!

### CODE REVIEW

File: module-2/backend/week_three.md [hunk 1]
```diff
@@ -10,14 +10,47 @@ Note: When you're done, submit a PR with a reflection in the comments about how
...
+  * ApplicationController
+  
 4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
+  * `resources :horses, only: [:show]`
+  
 5. What rake task is useful when looking at routes, and what information does it give you?
+  * `rake routes` returns a list of HTTP verb and controller acion pairs, the routes URL path, and a path/url helper method name.
+
 6. What is an example of a route helper? When would you use them?
+  * E.g., `new_horses_path`. Anytime you find yourself typing in a url string, it's probably (definitely?) time to use a path helper. It's a method under the Rails hood that determines the url based on convention.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: 👍

File: module-2/backend/week_three.md [hunk 2]
```diff
@@ -10,14 +10,47 @@ Note: When you're done, submit a PR with a reflection in the comments about how
...
+
 7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
+  * `_url` is more like an absolute path (full url), whereas `_path` is more of a path relative to the root path.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: 🎉

File: module-2/backend/week_three.md [hunk 3]
```diff
@@ -10,14 +10,47 @@ Note: When you're done, submit a PR with a reflection in the comments about how
...
+
 8. What are strong params and why are the necessary?
+  * Strong params are a requirement fo saving objects in a Rails database. They ensure that no extra/malicious information can be submitted into the database, only the attribute title included in the `#permit()` method will be processed for instantiation.
+
 9. What role does `form_for` play in helping us create our forms?
+  * This Rails helper 'Ruby-fies' an HTML form in a way that is more readable and clear w/r/t what is being made and where it should be saved. The form collects the data that will eventually be used to save as a new object's attributes.
+  
 10. How does `form_for` know where to submit the user's input?
+  * It reads the type of instance variable passed to its argument(s). If am object shell comes in (e.g., Horse.new), it knows to POST to `/horses`. If the object is already a Horse object, it knows to PUT to `horses/:id`.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: 🎉

File: module-2/backend/week_three.md [hunk 4]
```diff
@@ -10,14 +10,47 @@ Note: When you're done, submit a PR with a reflection in the comments about how
...
+    
+    <%= f.label :breed %>
+    <%= f.text_field :breed %>
+    
+    <%= f.submit %>
+  <% end %>
+  ```
+  
 12. Why do we want to validate our models?
+  * This adds a level of cleanliness to our database, further ensuring that each object has its necessary information and datatypes, where appropriate.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: 👍

**Comment by tmikeschu** [comment]
I added week four to this PR, just a heads up!

**Pull request closed without merging**